xcode/Subconscious/Shared/Models/Petname.swift

```diff
@@ -68,6 +68,13 @@ public struct Petname:
         self.verbatim = description
     }
     
+    /// Join a list of petnames into a dotted string, i.e. [foo, bar, baz] -> foo.bar.baz
+    /// Names are joined in order of their appearance in `petnames`
```

Comment:
I was tempted to call `.reversed()` on the input here, because the order of navigation in the app is reversed to the order that petnames will appear... 

i.e. `ben` -> `gordon` yields `gordon.ben`

But that feels like a hidden assumption. I'm leaving it up to the caller to decide on order for now.

Reply:
We should write some tests covering typical and pathological parsing cases. Off the top of my head.

```swift
XCTAssertNotNil(Slashlink("@bob"))
XCTAssertNotNil(Slashlink("@alice.bob"))
XCTAssertNotNil(Slashlink("@dan.charlie.bob.alice"))
XCTAssertNotNil(Slashlink("@bob-foo.alice-foo"))
XCTAssertNotNil(Slashlink("@bob_foo.alice_foo"))
XCTAssertNil(Slashlink("@eve...alice"))
XCTAssertNil(Slashlink("@.eve.alice"))
XCTAssertNil(Slashlink("@alice.eve."))
XCTAssertNil(Slashlink("@@eve.alice"))
XCTAssertNil(Slashlink("@eve@alice"))
XCTAssertNil(Slashlink("@eve//foo"))
XCTAssertNil(Slashlink("@eve..alice/foo"))
XCTAssertNil(Slashlink("@eve.evelyn/foo//bar"))
```

Reply:
Just added to the Slashlink parser in Noosphere, FWIW names are stored in written-order: https://github.com/subconsciousnetwork/subtext/pull/49/files

Reply:
Awesome, written order it is 👍